osu.Game.Rulesets.Catch/Judgements/CatchJudgementResult.cs

```diff
@@ -20,6 +21,8 @@ public class CatchJudgementResult : JudgementResult
         /// </summary>
         public bool CatcherHyperDash;
 
+        public CaughtObjectEntry CaughtObjectEntry;
```

Comment:
Not 100% happy with the layering/separation of concerns here... Should the judgement result really be aware of when/how it was caught via the entry? Maybe this would be better suited to the HO itself? Or be tracked somewhere external entirely, as I think was the case (dictionaries in `CaughtObjectContainer`)?

Reply:
A caught object is a result of a judgment (catching) so maintaining the caught object entry makes sense to me.
I cannot be a property of the HO because a hit object can be associated with multiple judgment results (not currently supported in `Playfield`, but used in tests, and I think it makes sense conceptually).
A caught object could be associated with `HitObjectLifetimeEntry` but there is no code path to modify this object before #11262.

Reply:
Do you have a preview or other loose branch of how this would look with that mentioned pull? That might provide better rationale for having that merged.

Reply:
I wrote a prototype of associating `CaughtObjectEntry` with `HitObjectLifetimeEntry` <https://github.com/ekrctb/osu/commit/392d048fe2e73236262427dc5bf0667b9b0ad31a>.
I think this way is not great. It need to expose `LifetimeEntry` in DHO to make it accessible.